include/realtime_tools/realtime_box_best_effort.h

```diff
@@ -0,0 +1,220 @@
+class RealtimeBoxBestEffort
+{
+  static_assert(
+    std::is_same_v<mutex_type, std::mutex> || std::is_same_v<mutex_type, std::recursive_mutex>);
+  static_assert(std::is_copy_constructible_v<T>, "Passed type must be copy constructible");
+
+public:
+  using mutex_t = mutex_type;
+  using type = T;
+  // Provide various constructors
+  constexpr explicit RealtimeBoxBestEffort(const T & init = T{}) : value_(init) {}
+  constexpr explicit RealtimeBoxBestEffort(const T && init) : value_(std::move(init)) {}
+
+  // Only enabled for types that can be constructed from an initializer list
+  template <typename U = T>
+  constexpr RealtimeBoxBestEffort(
+    const std::initializer_list<U> & init,
+    std::enable_if_t<std::is_constructible_v<U, std::initializer_list>>)
+  : value_(init)
+  {
+  }
+
+  /**
+     * @brief set a new content with best effort
+     * @return false if mutex could not be locked
+     * @note disabled for pointer types
+    */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, bool> trySet(const T & value)
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return false;
+    }
+    value_ = value;
+    return true;
+  }
+  /**
+   * @brief access the content readable with best effort
+   * @return false if the mutex could not be locked
+   * @note only safe way to access pointer type content (rw)
+  */
+  bool trySet(const std::function<void(T &)> & func)
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return false;
+    }
+
+    func(value_);
+    return true;
+  }
+  /**
+     * @brief get the content with best effort
+     * @return std::nullopt if content could not be access, otherwise the content is returned
+    */
+  template <typename U = T>
+  [[nodiscard]] typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, std::optional<U>> tryGet() const
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return std::nullopt;
+    }
+    return value_;
+  }
+  /**
+   * @brief access the content (r) with best effort
+   * @return false if the mutex could not be locked
+   * @note only safe way to access pointer type content (r)
+  */
+  bool tryGet(const std::function<void(const T &)> & func)
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return false;
+    }
+
+    func(value_);
+    return true;
+  }
+
+  /**
+     * @brief set the content and wait until the mutex could be locked (RealtimeBox behaviour)
+     * @return true
+    */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, void> set(const T & value)
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    // cppcheck-suppress missingReturn
+    value_ = value;
+  }
+  /**
+   * @brief access the content (rw) and wait until the mutex could locked
+  */
```

Comment:
I think that this, as here with my proposal is, is a correct alignment.
```suggestion
   */
```